Load testing guidelines docs (#12581)
RasaHQ/rasa: docs/docs/monitoring/load-testing-guidelines.mdx

---
sidebar_label: Load Testing Guidelines
title: Load Testing Guidelines
description: |

>   Information about how best to scale up your bot to support more parallel user activity

Reviewer: Or we can say `to support an increased number of parallel user activity`

>   Information about how best to scale up your bot to support more parallel user activity

Reviewer 2: +1

>   Information about how best to scale up your bot to support more parallel user activity

Author: good catch, 'more' was a leftover from a different sentence I was trying out.

  and how you can use tracing to help debug issues.
---

:::note

> Our team is currently in the process of running more extensive performance related tests, more information will be added here as we progress.

Reviewer 2: ```suggestion
Our team is currently in the process of running additional performance-related tests. More information will be added here as we progress.
```

> :::

Author: Open to suggestions, this is to let people know that this is in progress.

## Overview

In the process to gather metrics on our system's ability to handle increased loads and users we have performed tests to evaluate the maximum number of concurrent users a Rasa chatbot can handle with certain machine configurations.

> In each test case we spawned the following number of concurrent users at peak concurrency using a spawn rate of 1000 users per second. In our tests we used Rasa's [Http-Api](https://rasa.com/docs/rasa/pages/http-api),

Reviewer: Can we add a definition first of what a spawn rate is, or link to some established glossary online?

Author: Linked a locust configuration page with a definition of spawn rate mentioned but nothing specific.

> In each test case we spawned the following number of concurrent users at peak concurrency using a spawn rate of 1000 users per second. In our tests we used Rasa's [Http-Api](https://rasa.com/docs/rasa/pages/http-api),

Reviewer: ```suggestion
In each test case we spawned the following number of concurrent users at peak concurrency using a spawn rate of 1000 users per second. In our tests we used the Rasa [HTTP-API](https://rasa.com/docs/rasa/pages/http-api),
```
I have a stylistic preference for uppercase 😄

> along with common opensource load testing tool such as [locust](https://locust.io/).

Reviewer: ```suggestion
and the [Locust](https://locust.io/). open source load testing tool.
```

> along with common opensource load testing tool such as [locust](https://locust.io/).

Reviewer 2: ```suggestion
along with a common open source load testing tool such as [locust](https://locust.io/).
```

|        Users             |               Configuration                        |
|--------------------------|----------------------------------------------------|
| Up to 50,000             |         6vCPU, 16 GB                               |

> | Up to 80,000             |         6vCPU, 16 GB but with almost 90% CPU usage |

Reviewer: Can we actually have CPU usage as a separate column actually for both cases.

Reviewer 2: Do we only have data for these two 'user counts/reference architectures'? Or am I missing something? Perhaps we can ask CSEs to give us the most common 'reference architectures' of our customers?

Author: @ancalita I can put CPU usage as a seperate column, I stuck to the format that was used [here](https://docs.gitlab.com/ee/administration/reference_architectures/1k_users.html)

Reviewer 2: okay, gotcha! but I wonder if this creates value for customers in this form if our examples are not very close to their user numbers. let's do this now and try to make it more valuable for next iteration!

Author: Already on it, plus hoping to get more ideas on the friday meeting with the team.
